xarray/core/treenode.py

```diff
@@ -44,6 +45,73 @@ def __init__(self, *pathsegments):
 Tree = TypeVar("Tree", bound="TreeNode")
 
 
+class Children(Mapping[str, Tree], Generic[Tree]):
+    """
+    Dictionary-like container for the immediate children of a single DataTree node.
+
+    This collection can be passed directly to the :py:class:`~xarray.DataTree` constructor via its `children` argument.
+    """
+
+    _treenode: Tree
+
+    # TODO add slots?
+    # __slots__ = ("_data",)
+
+    def __init__(self, treenode: Tree):
+        self._treenode = treenode
+
+    @property
+    def _names(self) -> list[str]:
+        return list(self._treenode._children.keys())
+
+    def __iter__(self) -> Iterator[str]:
+        return iter(self._names)
+
+    def __len__(self) -> int:
+        return len(self._names)
+
+    def __contains__(self, key: str) -> bool:
+        return key in self._names
+
+    def __repr__(self) -> str:
+        return "\n".join(["Children:"] + [f"    {name}" for name in self._names])
+
+    def __getitem__(self, key: str) -> Tree:
+        return self._treenode._children[key]
+
+    def __delitem__(self, key: str) -> None:
+        if key in self._names:
+            child = self._treenode._children[key]
+            del self._treenode._children[key]
+            child.orphan()
+        else:
+            raise KeyError(key)
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        self.update({key: value})
+
+    def update(self, other: Mapping[str, Tree]) -> None:
+        """Update with other child nodes."""
+
+        # TODO forbid strings with `/` in here?
+
+        if not len(other):
+            return
```

Comment:
It's slightly more canonical Python to avoid the call to `len()`:
```suggestion
        if not other:
            return
```